third_party/move/move-vm/runtime/src/runtime_type_checks.rs

```diff
@@ -386,6 +456,22 @@ impl RuntimeTypeCheck for FullRuntimeTypeCheck {
                 let field_ref_ty = ty_builder.create_ref_ty(field_ty, is_mut)?;
                 operand_stack.push_ty(field_ref_ty)?;
             },
+            Bytecode::PackClosure(fh_idx, mask) => {
+                let function =
+                    resolver.build_loaded_function_from_handle_and_ty_args(*fh_idx, vec![])?;
+                verify_pack_closure(resolver, operand_stack, &function, *mask)?;
+            },
+            Bytecode::PackClosureGeneric(fh_idx, mask) => {
+                let ty_args = resolver.instantiate_generic_function(
+                    Option::<&mut UnmeteredGasMeter>::None, // don't charge for paranoid mode
+                    *fh_idx,
+                    ty_args,
+                )?;
+                let function = resolver
```

Comment:
What are your thoughts on merging this in interpreter loop? also cc @runtian-zhou 

We build types and access closures, but this info is shared, I do not think having switch over pre-processing checks, switch over execution, switch over post-processing checks is so much better than clear structure in interpreter:
```
Bytecode::instruction) => {
  // Paranoid pre-checks: None
  // Gas metering:
  gas_meter.charge(...)?

  // Execution:
  ...

  // Paranoid post-checks:
  ....
}
```

Mote: irrelevant for this PR, but should we do this as a follow up?

Reply:
As we discussed the other day, one could build a visitor API which gets parameters for shared derived info. (Though ideally the parameters would be some kind of lazy lambda -- compute info only if needed but only once.) However, I feel its not worth the effort for the v1 VM. FWIW, Runtian had these checks before in the interpreter, Zi moved it here.